src/client.js

```diff
@@ -71,13 +73,20 @@ let getRegistration = async function() {
   }
 };
 
-let pushClient = {
-  async subscribe({workerUrl=WORKER_URL, endpoint=null} = {}) {
-    // Check for support
-    if (!this.supported()) {
-      throw new SubscriptionFailedError('not supported');
+class PushClient {
+  constructor({endpointUrl=null, userId=null, workerUrl=WORKER_URL,
+      scope=SCOPE}) {
+    if (!PushClient.supported()) {
+      throw new Error('Your browser does not support the web push API');
     }
 
+    this.endpoint = endpointUrl ? new Endpoint(endpointUrl) : null;
+    this.userId = userId;
```

Comment:
What is the use case for the userID here?

If its to handle something along the lines of authenticating the user with the developers service I can't help but feel like this is overextending the library, I would stick with what the library needs to focus on - nothing more. Add it when it's request.